config_parser.cc

```diff
@@ -198,18 +200,28 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
       config_stack.top()->statements_.back().get()->child_block_.reset(
           new_config);
       config_stack.push(new_config);
+      block_lvl++;
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
-      if (last_token_type != TOKEN_TYPE_STATEMENT_END) {
+      // in addition to checking statement_end, also need to check block_end
+      if (last_token_type != TOKEN_TYPE_STATEMENT_END && last_token_type != TOKEN_TYPE_END_BLOCK) {
```

Comment:
`last_token_type` needs to be checked within the `else if` to make sure that nested curly braces are valid. Or else the following will fail.

```C++
EXPECT_TRUE(ParseString("#test\n\
                              server {\n\
                                server {listen 80;}\n\
                                     }")) 
      << "Error: failed to parse doubly nested braces + comment correctly";
```